string-manipulation-3/src/title-case.js

```diff
@@ -0,0 +1,50 @@
+/* eslint-disable no-unused-vars */
+function titleCase(title) {
+  var titleArray = title.split(' ');
+  var newArray = [];
+  var newIndividuals = [];
+  for (var i = 0; i < titleArray.length; i++) {
+    var current = titleArray[i];
+    if (current.length >= 4) {
+      if (current === 'javascript' || current === 'Javascript:' || current === 'javascript:') {
+        var j = current[0].toUpperCase();
+        var middle = current.substring(1, 4).toLowerCase();
+        var s = current[4].toUpperCase();
+        var end = current.substring(5).toLowerCase();
+        current = j + middle + s + end;
+        newArray.push(current);
+      } else {
+        var firstLetter = current[0].toUpperCase();
+        var rest = current.substring(1);
+        var word = firstLetter + rest;
+        newArray.push(word);
+      }
+    } else if (current === 'api') {
+      current = current.toUpperCase();
+      newArray.push(current);
+    } else if (current === 'web') {
+      current = current[0].toUpperCase() + current[1] + current[2];
+      newArray.push(current);
+    } else {
+      newArray.push(current);
+    }
+  }
+  var newString = newArray.join(' ');
+  var individuals = newString.split('');
+  for (var x = 0; x < individuals.length; x++) {
+    if (individuals[x] === ':') {
+      var upperCased = individuals[x + 2].toUpperCase();
+      newIndividuals.push(individuals[x] + ' ' + upperCased);
+      x += 3;
+    }
+    if (individuals[x] === '-') {
+      var capital = individuals[x + 1].toUpperCase();
+      newIndividuals.push(individuals[x] + capital);
+      x += 1;
+    } else {
+      newIndividuals.push(individuals[x]);
+    }
+  }
+  var finalString = newIndividuals.join('');
+  return finalString;
```

Comment:
Ok, this code works for the current inputs, but it's pretty hard to follow. Since the problems statement outlines some rules, try to incorporate them directly in your code. Whenever possible, use things from the ["problem domain"](https://softwareengineering.stackexchange.com/questions/125926/what-is-problem-domain).

For reference, here is the problem statement:

Rules Summary
- Capitalize the first word of the title and of any subtitle.
- Capitalize all “major” words (nouns, verbs, adjectives, adverbs, and pronouns) in the title, including the second part of hyphenated major words (e.g., Self-Report not Self-report).
- Capitalize all words of four letters or more.

This boils down to using lowercase only for “minor” words of three letters or fewer, namely, for conjunctions (words like and, or, nor, and but), articles (the words a, an, and the), and prepositions (words like as, at, by, for, in, of, on, per, and to), as long as they aren’t the first word in a title or subtitle.

Special Cases
- JavaScript is always cased JavaScript.
- API is a "known" acronym.

Code is about communication, not only with a machine, but other programmers too. You want to communicate your intent as much as possible.

Try writing yourself some helper functions and choosing variable names that come from the rules:
- `isMinorWord(word)` - tells you whether or not (boolean) a `word` is one of the minor words.
- `capitalizeWord(word)` - you've already done this in the past; returns you the `word` capitalized. this could be spiced up to also handle hyphenated words 💡 
- `isKnownWord(word)` - let's you know if the `word` is one of the special cases (boolean)
- `capitalizeKnown(word)` - this one is a nice cheat; takes something like `'api'` or `'javascript'` and just returns the correct casing